mitiq/calibration/calibrator.py

```diff
@@ -252,34 +254,50 @@ def best_strategy(self) -> Strategy:
     def execute_with_mitigation(
         self,
         circuit: QPROGRAM,
-        expval_executor: Union[Executor, Callable[[QPROGRAM], QuantumResult]],
+        expval_executor: Optional[
+            Union[Executor, Callable[[QPROGRAM], QuantumResult]]
+        ] = None,
```

Comment:
Having `None` here as default is dangerous since it is easy to miss the fact that there is a hidden default bitstring under the hood. Having `expval_executor` as a required argument seems safer to me.